btsieve/src/lib.rs

```diff
@@ -38,3 +40,27 @@ impl PartialOrd for QueryMatch {
         self.0.partial_cmp(&other.0)
     }
 }
+
+struct BlockchainDAG<T, V> {
+    nodes: Vec<T>,
+    vertices: Vec<V>,
+}
+
+impl Default for Bitcoin {
+    fn default() -> Self {
+        Self(BlockchainDAG {
+            nodes: Vec::new(),
+            vertices: Vec::new(),
+        })
+    }
+}
+
+struct Bitcoin(BlockchainDAG<MinedBlock, (Sha256dHash, Sha256dHash)>);
+
+struct Ethereum;
+
+trait AddBlock<T> {
```

Comment:
I do not know typical Rust naming for traits but this name does not seem to match the trait functions. Suggest `Blockchain`.

Reply:
Also perhaps `default` should be in the trait (if that is how Rust likes it)?

Reply:
True, at the beginning there was only one function --> I'll update it.